### PR TITLE
[front] - fix(retrieval): allow nullable retrivalActionId on RetrievalDocument

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

Updated the foreign key constraint on RetrievalDocument to allow null values, changing deletion policy to set it null instead of cascading delete.

## Deploy Plan

- Deploy front
- Apply migration

### CODE REVIEW

File: front/lib/models/assistant/actions/retrieval.ts [hunk 1]
```diff
@@ -326,11 +326,11 @@ RetrievalDocument.init(
 );
 
 AgentRetrievalAction.hasMany(RetrievalDocument, {
```
Comment (flvndvd) [inline front/lib/models/assistant/actions/retrieval.ts:328]: I guess we need to update the type line 273?

File: front/migrations/db/migration_124.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+-- Drop existing foreign key constraint
+ALTER TABLE retrieval_documents
+DROP CONSTRAINT retrieval_documents_retrievalactionid_fkey;
```
Comment (flvndvd) [inline front/migrations/db/migration_124.sql:3]: Constraint name is `"retrieval_documents_retrievalActionId_fkey"`

File: front/migrations/db/migration_124.sql [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+-- Allow null and add new foreign key with SET NULL
+ALTER TABLE retrieval_documents
+ALTER COLUMN "retrievalActionId" DROP NOT NULL,
+ADD CONSTRAINT retrieval_documents_retrievalActionId_fkey
```
Comment (flvndvd) [inline front/migrations/db/migration_124.sql:8]: You need the double quotes to preserve the cases.

File: front/migrations/db/migration_124.sql (hunk 2 above)
Comment (JulesBelveze) [reply front/migrations/db/migration_124.sql:3]: Indeed, weird for some reason I had it lowercased on my local 🤷🏼‍♂️

File: front/lib/models/assistant/actions/retrieval.ts (hunk 1 above)
Comment (JulesBelveze) [reply front/lib/models/assistant/actions/retrieval.ts:328]: 👍🏼

File: front/migrations/db/migration_124.sql [hunk 4]
```diff
@@ -0,0 +1,11 @@
+-- Drop existing foreign key constraint
+ALTER TABLE retrieval_documents
+DROP CONSTRAINT "retrieval_documents_retrievalActionid_fkey";
```
Comment (flvndvd) [inline front/migrations/db/migration_124.sql:3]: Still not the right name, in production we have: `retrieval_documents_retrievalActionId_fkey`.

Comment (JulesBelveze) [reply front/migrations/db/migration_124.sql:3]: My fault, I copied from my local. Copied it from prodbox, should be fine now 🙏🏼

**Review by flvndvd — APPROVED** [review]

**Pull request merged**